src/EVM/Expr.hs

```diff
@@ -478,6 +454,54 @@ minLength bufEnv = go 0
       b <- Map.lookup a bufEnv
       go l b
 
+-- returns the largest prefix that is guaranteed to be concrete (if one exists)
+concretePrefix :: Expr Buf -> Maybe (Vector (Expr Byte))
```

Comment:
this should just return `Maybe (Vector Byte)` since they should all be concrete. Even better would be a `ByteString`, but the fast creation routines for `ByteString` all seem to expect a concrete size in advance (which we don't always know).